### ISSUE TITLE
OEM kernel variation for better modern hardware support

### ISSUE THREAD

**Issue by ibnesayeed** [elementary/os#434]
There are some recent laptops shipping with hardware that is not well supported by the Linux kernel 5.4 (shipped in Ubuntu 20.04). Recently, I tried installing the Early Access version of elementary OS 6 on Dell XPS 17 (9700), but the audio did not work. However, [I got it working by using the OEM kernel in my build](https://github.com/elementary/switchboard-plug-sound/issues/181), which has necessary drivers included.

With that said, I propose we go for one of the following options for a better user experience:

* Ship the main build with the OEM kernel, which usually has better hardware support
* Provide an alternate build with the OEM kernel, in addition to the main build with the generic kernel
* Release the build with only the generic kernel, document the process of installing the OEM one for those have driver issues

**Comment by danrabbit** [comment]
My understanding is that the OEM kernel is effectively a staging area for the HWE kernel and has a shorter support lifecycle. According to [this page](https://wiki.ubuntu.com/Kernel/OEMKernel) It may also include workarounds that wouldn't be accepted by the SRU process and there's no guarantee that it won't cause regressions on hardware that is outside of the Ubuntu certification process.

**Comment by ibnesayeed** [comment]
I did not explore Linux kernel side of affairs and how Ubuntu handles things, until recently. If I understood things correctly, then the support that was added in the OEM (which made the audio work on XPS 17), should become part of their generic kernel in the next release (which seems to be January 25, i.e., a week from now). Though, I am not sure when was the fix added to the OEM kernel. If it was added earlier, then it should have been merged in the last SRU cycle (released on January 4).

I think what it all boils down to is, if the journey from the oem to the generic kernel is quick enough then we need not to worry about it too much. My fear was, if Ubuntu wouldn't backport these drivers to the generic kernel of the 20.04 series, then elementary OS 6 will suffer from the lack of support for the next 2-3 years until version 7 (based on Ubuntu 22.04 is out). However, after reading their documentation, I feel that it is likely not going to be the case.
